### PR TITLE
[cli] add docs for `detach` command

### PR DESCRIPTION
**Pull request by abtink**

### CODE REVIEW

File: src/cli/README.md [hunk 1]
```diff
@@ -1061,6 +1062,25 @@ Set the minimal delay timer (in seconds).
...
+
+```bash
+> detach
+Finished detaching
+Done
+```
+
+### detach async
+
+Start the graceful detach process similar to `detach` command without blocking and waiting for the callback indicating that detach is finished.
```
Comment (Vyrastas) [inline src/cli/README.md:1077]: to the `detach` command

File: src/cli/cli.cpp [hunk 2]
```diff
@@ -2443,25 +2443,57 @@ template <> otError Interpreter::Process<Cmd("delaytimermin")>(Arg aArgs[])
...
     otError error = OT_ERROR_NONE;
 
+    /**
+     * @cli detach async
+     * @code
+     * detach async
+     * Done
+     * @endcode
+     * @par
+     * Start the graceful detach process similar to `detach` command without blocking and waiting for the callback
```
Comment (Vyrastas) [inline src/cli/cli.cpp:2469]: to the `detach` command

File: src/cli/cli.cpp [hunk 3]
```diff
@@ -2443,25 +2443,57 @@ template <> otError Interpreter::Process<Cmd("delaytimermin")>(Arg aArgs[])
...
+     * indicating that detach is finished.
+     */
```
Comment (Vyrastas) [inline src/cli/cli.cpp:2471]: Can you add a csa tag to add a link for the detach command?

`* @csa{detach}`

File: src/cli/cli.cpp [hunk 4]
```diff
@@ -2443,25 +2443,57 @@ template <> otError Interpreter::Process<Cmd("delaytimermin")>(Arg aArgs[])
...
+ * Start the graceful detach process by first notifying other nodes (sending Address Release if acting as a router, or
+ * setting Child Timeout value to zero on parent if acting as a child) and then stopping Thread protocol operation.
+ * @sa otThreadDetachGracefully
+ */
 template <> otError Interpreter::Process<Cmd("detach")>(Arg aArgs[])
 {
     otError error = OT_ERROR_NONE;
 
+    /**
+     * @cli detach async
```
Comment (Vyrastas) [inline src/cli/cli.cpp:2463]: This one uses the otThreadDetachGracefully API too, why not `@sa` to it as well?

**Review by Vyrastas — CHANGES_REQUESTED** [review]

File: src/cli/cli.cpp (hunk 4 above)
Comment (abtink) [reply src/cli/cli.cpp:2462]: Thanks @Vyrastas. New push contains all the changes.

**Review by Vyrastas — APPROVED** [review]

**Pull request merged**